draft-ietf-tls-external-psk-importer.md

```diff
@@ -155,47 +156,47 @@ PSK is being imported. For example, TLS 1.3 {{!RFC8446}} and QUICv1 {{!QUIC=I-D.
    ipskx = HKDF-Expand-Label(epskx, "derived psk",
                              Hash(ImportedIdentity), L)
 ~~~
 
-L is corresponds to the KDF output length of ImportedIdentity.target_kdf as defined in {{IANA}}.
+L corresponds to the KDF output length of ImportedIdentity.target_kdf as defined in {{IANA}}.
 For hash-based KDFs, such as HKDF_SHA256(0x0001), this is the length of the hash function
 output, i.e., 32 octets. This is required for the IPSK to be of length suitable for supported
 ciphersuites.
 
-The identity of 'ipskx' as sent on the wire is ImportedIdentity, i.e., the serialized content
+The identity of `ipskx` as sent on the wire is ImportedIdentity, i.e., the serialized content
 of ImportedIdentity is used as the content of PskIdentity.identity in the PSK extension.
+The corresponding TLS 1.3 binder key is `ipskx`.
 
 The hash function used for HKDF {{!RFC5869}} is that which is associated with the EPSK.
 It is not the hash function associated with ImportedIdentity.target_kdf. If no hash function
 is specified, SHA-256 MUST be used. Diversifying EPSK by ImportedIdentity.target_kdf ensures
 that an IPSK is only used as input keying material to at most one KDF, thus satisfying
-the requirements in {{!RFC8446}}.
-
-Endpoints generate a compatible ipskx for each target ciphersuite they offer. For example, importing a
-key for TLS_AES_128_GCM_SHA256 and TLS_AES_256_GCM_SHA384 would yield two PSKs, one for HKDF-SHA256 and
-another for HKDF-SHA384. In contrast, if TLS_AES_128_GCM_SHA256 and TLS_CHACHA20_POLY1305_SHA256
-are supported, only one derived key is necessary.
+the requirements in {{!RFC8446}}. See {{security-considerations}} for more details.
 
-The resulting IPSK base key 'ipskx' is then used as the binder key in TLS 1.3 with identity
-ImportedIdentity. With knowledge of the supported KDFs, one may import PSKs before the start
-of a connection.
+Endpoints SHOULD generate a compatible `ipskx` for each target ciphersuite they offer.
+For example, importing a key for TLS_AES_128_GCM_SHA256 and TLS_AES_256_GCM_SHA384 would
+yield two PSKs, one for HKDF-SHA256 and another for HKDF-SHA384. In contrast, if
+TLS_AES_128_GCM_SHA256 and TLS_CHACHA20_POLY1305_SHA256 are supported, only one
+derived key is necessary.
 
-EPSKs may be imported for early data use if they are bound to protocol settings and configurations that would
-otherwise be required for early data with normal (ticket-based PSK) resumption. Minimally, that means ALPN,
-QUIC transport settings, etc., must be provisioned alongside these EPSKs.
+EPSKs may be imported before the start of a connection if the target KDFs, protocols, and
+context string(s) are known a priori. EPSKs may also be imported for early data use
+if they are bound to protocol settings and configurations that would otherwise be
+required for early data with normal (ticket-based PSK) resumption. Minimally, that
+means ALPN, QUIC transport settings, etc., must be provisioned alongside these EPSKs.
```

Comment:
This is an existing sentence, but the word "minimally" here is interesting given PSKs meant for use with non-QUIC connections don't need QUIC transport settings (nitpick: I think QUIC calls them "transport parameters").

Reply:
Good catch -- fixed!